pyproject.toml

```diff
@@ -6,23 +6,21 @@ authors = ["Team Microdata"]
 license = "Apache 2.0"
 
 [tool.poetry.dependencies]
-python = "^3.10"
-Flask = "^2.3.2"
-Flask-Pydantic = "^0.9.0"
+python = "^3.12"
+Flask = "^3.0.0"
+Flask-Pydantic = "^0.12.0"
 pymongo = "^4.0.2"
-pydantic = "^1.9.0"
+pydantic = "^2.6.1"
 PyJWT = "^2.4.0"
 cryptography = "^42.0.4"
-gunicorn = "^20.1.0"
+gunicorn = "^21.2.0"
 json-logging = "^1.3.0"
 tomlkit = "^0.10.1"
-pyhumps = "^3.7.2"
 
 [tool.poetry.group.dev.dependencies]
-mypy = "^1.3.0"
-pytest = "^7.1.1"
+pytest = "^8.0.1"
 pytest-dotenv = "^0.5.2"
-pytest-cov = "^3.0.0"
+pytest-cov = "^4.1.0"
 pytest-mock = "^3.7.0"
 testcontainers = {extras = ["mongodb"], version = "^3.7.0"}
 ruff = "^0.1.7"
```

Comment:
upgrade ruff to `ruff = "^0.2.0"` like in [microdata-tools](https://github.com/statisticsnorway/microdata-tools) (if we are all okay with using ruff for formatting 👍) @pawbu @linnbjornerud